tests/integration/graviton_abi_test.py

```diff
@@ -541,3 +544,54 @@ def test_conditional_factorial(version: int):
     assert inspector.error(), inspector.report(
         args, f"FAILED: should error for {n=}", row=n + 1
     )
+
+
+xs = list(map(pynum_to_int65tuple, (-1, 1, 3, 5, 7)))
+int65_unary_function_inputs = [(x,) for x in xs]
+int65_binary_function_inputs = list(product(xs, xs))
+zs = int65_binary_function_inputs
+complex130_unary_function_inputs = [(z,) for z in zs]
+complex130_binary_function_inputs = list(product(zs, zs))
+VERSIONING_CASES = [
+    (int65_minus_cond, int65_binary_function_inputs),
+    (int65_sub, int65_binary_function_inputs),
+    (int65_mult, int65_binary_function_inputs),
+    (int65_negate, int65_unary_function_inputs),
+    (int65_add, int65_binary_function_inputs),
+    (complex130_add, complex130_binary_function_inputs),
+    (complex130_mult, complex130_binary_function_inputs),
+    (complex130_real, complex130_unary_function_inputs),
+    (complex130_imag, complex130_unary_function_inputs),
+    (complex130_conjugate, complex130_unary_function_inputs),
+    (complex130_norm_squared, complex130_unary_function_inputs),
+]
+
+APP_PREDICATES = {
+    DRProp.lastLog: PredicateKind.IdenticalPair,
+    DRProp.status: PredicateKind.IdenticalPair,
+    DRProp.error: PredicateKind.IdenticalPair,
+    DRProp.lastMessage: PredicateKind.IdenticalPair,
+}
+LSIG_PREDICATES = {
+    DRProp.status: PredicateKind.IdenticalPair,
+    DRProp.error: PredicateKind.IdenticalPair,
+    DRProp.lastMessage: PredicateKind.IdenticalPair,
+}
+min_version = 5
+
+
+@pytest.mark.parametrize("subroutine, inputs", VERSIONING_CASES)
+@pytest.mark.parametrize("mode", pt.Mode)
+@pytest.mark.parametrize("version", range(min_version + 1, pt.MAX_PROGRAM_VERSION + 1))
+def test_identical_functionality(subroutine, inputs, mode, version):
```

Comment:
I don't recommend merging as is. It's nice that it ran once and passed (hopefully). But it's too slow to justify keeping around for posterity. However, if we find lighter functions (and hopefully that are able to shed more light) we should be able to run a similar test that runs 100X faster.

Reply:
oh boy... that takes too much time, do we want to test only one version pre frame and one version post frame? For we are testing version range [5, 8], so that should cut to 50% of time, but the rest of optimization is non-trivial...

Reply:
That makes sense to me!